raster/r.info/tests/test_r_info.py

```diff
@@ -0,0 +1,160 @@
+"""
+Name:       r.info test
+Purpose:    Tests r.info and its flags/options.
+
+Author:     Sunveer Singh (Google Code-in 2017), Kriti Birda
+Copyright:  (C) 2017 by Sunveer Singh, Kriti Birda and the GRASS Development Team
+Licence:    This program is free software under the GNU General Public
+                License (>=v2). Read the file COPYING that comes with GRASS
+                for details.
+"""
+
+import json
+
+import grass.script as gs
+from grass.gunittest.checkers import text_to_keyvalue
+import pytest
+
+json_format_expected_no_stats = {
+    "north": 100,
+    "south": 0,
+    "east": 100,
+    "west": 0,
+    "nsres": 1,
+    "ewres": 1,
+    "rows": 100,
+    "cols": 100,
+    "min": 1,
+    "max": 3,
+    "cells": 10000,
+    "datatype": "CELL",
+    "ncats": 3,
+    "map": "landuse",
+    "maptype": "raster",
+    "mapset": "PERMANENT",
+    "title": "Sample Land-Use",
+    "timestamp": None,
+    "units": None,
+    "vdatum": None,
+    "semantic_label": None,
+    "source1": "",
+    "source2": "",
+    "description": "generated by r.mapcalc",
+    "comments": "if(row() < 33, 1, if(row() < 66, 2, 3))",
+}
+
+
+def assert_rinfo_output(expected_str, received_str):
+    expected = dict(**text_to_keyvalue(expected_str, sep="=", skip_empty=True))
+    received = dict(**text_to_keyvalue(received_str, sep="=", skip_empty=True))
+    assert expected == received
+
+
+def assert_rinfo_json_output(expected, received):
+    # the following fields vary with the Grass sample data's path
+    # therefore only check for their presence in the JSON output
+    # and not exact values
+    remove_fields = ["location", "project", "database", "date", "creator"]
+    for field in remove_fields:
+        assert field in received
+        received.pop(field)
```

Comment:
I'm not sure what to do with this. I didn't find a good way how to do this in pytest with a one-liner. `unittest.mock.ANY` seems to be usable with pytest as a dict value when pytest-mock is installed. That may be little too much for this PR, but good to keep in mind.